MEMMODEL_CONSTRAINTS.md

```diff
@@ -103,6 +103,19 @@ may perceive in that statement.)
   for memory used for atomics.  The situation is somewhat better in
   well-typed C++ but reality is complicated.
 
+## Implementation targeting C and C++ or LLVM
+
+* _Targeting C or C++_: it should be possible to implement
+  SharedArrayBuffer using C or C++.  The C/C++11 memory model allows
+  accesses with a number of orderings, and there should be a mapping
+  from SharedArrayBuffer accesses to C/C++11 orderings. Racy
+  SharedArrayBuffer accesses are required to produce defined values,
+  so cannot be implemented using C/C++ non-atomics.
+* _Targeting LLVM_: implementations may also decide to target
+  LLVM, which may provide for more optimization than targeting C/C++
+  and compiling to LLVM. In particular, LLVM allows unordered access,
+  which does not require per-variable sequential consistency.
```

Comment:
> Implementing SAB -> LLVM IR -> x86/ARM/etc should totally be possible. As you point out it's not clear what restrictions the SAB spec is currently putting on what LLVM IR you can generate.

Yay, we seem to be on the same page now!

> Now that I understand what you're trying to do (SAB -> LLVM IR -> ...) I think mentioning C++ is misleading.

OK, I can remove it and just talk about LLVM. Should I move the edits to being back under hardware if we're just talking about LLVM? Or would you still prefer a separate section?

> That wasn't clear at all to me. I'd appreciate it if you could clarify the wording.

Will do.
 
> I wouldn't mention it if it's a moving target. Further, if Rust's recommended parallelism datastructures are shared-nothing then this discussion is kinda misleading. I'd at least like a link referencing the Rust memory model you have in mind.

I'll remove Rust. If you're interested, the bit of Rust's stdlib that's of interest is the atomics, which are just a wrapper around LLVM's atomics: https://doc.rust-lang.org/std/sync/atomic/index.html